cmd/git/main.go

```diff
@@ -295,13 +293,28 @@ func clone(ctx context.Context) error {
 	if flagValues.depth > 0 {
 		submoduleArgs = append(submoduleArgs, "--depth", fmt.Sprintf("%d", flagValues.depth))
 	}
-	_, err := git(ctx, submoduleArgs...)
-	return err
+
+	if _, err := git(ctx, submoduleArgs...); err != nil {
+		return err
+	}
+
+	head, err := ioutil.ReadFile(filepath.Join(flagValues.target, ".git", "HEAD"))
+	if err != nil {
+		return err
+	}
+
+	log.Printf("Successfully loaded %s (%s) into %s\n",
+		flagValues.url,
+		strings.TrimRight(string(head), "\n"),
+		flagValues.target,
+	)
```

Comment:
Not sure if we should rely on Git's internal directory structure. What about if we use the user-provided revision value, or fallback to the cloned branch name in case no revision was provided?

```suggestion
	revision := flagValues.revision
	if revision == "" {
		if revision, err = git(ctx, "-C", flagValues.target, "rev-parse", "--abbrev-ref", "HEAD"); err != nil {
			return err
		}
		revision = strings.TrimRight(revision, "\n")
	}

	log.Printf("Successfully loaded %s (%s) into %s\n",
		flagValues.url,
		revision,
		flagValues.target,
	)
```

(Quickly hacked together, did not actually try it out.)

Reply:
Yeah, I was playing around with the `git` command, but was not happy with the output. For example, when you are in detached HEAD state, which I believe we can have if a user specifies a commit SHA, the output would be:
```sh
$ git rev-parse --abbrev-ref HEAD
HEAD
```

So I was looking for an option to have a readable reference for detached HEAD and on branch and the only option that I was able to find was to rely on the Git metadata itself. With Git, I think we have a reliable contract.

Reply:
Yeah, totally agreeing on the detached state. That's why I proposed to only call `git rev-parse` when the user did NOT specify a revision. In such a case, the default branch gets cloned and we should be good, I think? I added a comment to the code.